src/python/pants/engine/scheduler.py

```diff
@@ -465,8 +471,10 @@ def schedule(self, execution_request):
 
       if self._scheduler.visualize_to_dir() is not None:
         name = 'run.{}.dot'.format(self._run_count)
+        rule_graph_name = 'rule_graph.{}.dot'.format(self._run_count)
```

Comment:
This isn't going to change run-over-run, so just doing it once during initialization should be equivalent? And no need to inclue the `run_count` in the name.